notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -0,0 +1,1127 @@
+        id numbers in the long format dataset.
+    obs_id_col: string
+        Name of the column containing the observation
+        id numbers in the long format dataset.
+    number_alts: int
+        Number of alternatives in the long format
+        dataset.
+    spec_dic: dictionary
+        Dictionary of the model specification.
+    names_dic: dictionary
+        Dictionary of the alternative names.
+    init_betas: list
+        List of the initial betas for the model
+        from which the choices should be simulated.
+    
+    Returns
+    -------
+    DataFrame object with the simulated choices column
+    added as 'sim_choice'
+    """
+    # Declare the simulated choice column name
+    sim_choice_col = 'sim_choice'
+
+    # Make a copy of the data
+    data = copy.deepcopy(long_data)
+    # Functions to generate the probabilities for each alternative
+    design_matrix, names = create_design_matrix(df_long=data,
+                                                specification_dict=spec_dic,
+                                                names_dict=names_dic,
+                                                alternative_id_col=alt_id_col)
+    mapping_matrix = create_mapping_matrix(df_long=data,
+                                           observation_id_col=obs_id_col)
+    probabilities = calculate_probabilities(betas=initial_betas,
+                                            design_matrix=design_matrix,
+                                            mapping_matrix=mapping_matrix)
+
+    data['probabilities'] = probabilities
+    data['cum_sum'] = 0
+    data['sim_choice'] = 0
+
+    for observation in data['observation_id'].unique():
+        probs_sum = data[long_data.observation_id == observation]['probabilities'].cumsum()
+        data.loc[data['observation_id'] == observation, 'cum_sum'] = probs_sum
+
+    observation_id_list = list(data.observation_id.unique())
+    u_random = np.random.uniform(size=len(data['observation_id'].unique()))
+
+    for u, obs in zip(u_random, observation_id_list):
+        data_sample = data[data['observation_id'] == obs]
+        sorted_list = sorted(list(data_sample['mode_id'].unique()))
+        choices = dict.fromkeys(sorted_list, 0)
+        for alt in sorted_list:
+            choices[alt] = np.where(u <= data_sample[data_sample['mode_id']
+                                                     == alt]
+                                    ['cum_sum'], 1, 0).item()
+            if choices[alt] == 1:
+                break
+        data.loc[data.observation_id == obs, sim_choice_col] = \
+            data['mode_id'].map(choices)
+    return data
+```
+
+# Example using bike data 
+
+
+## Data Ingestion and Exploration 
+
+```python
+# Create a variable for the path to the long format data for
+# the multinomial choice model
+PATH = '/Users/mobouzaghrane/Documents/GitHub/tr_b_causal_2020/'\
+        'data/raw/spring_2016_all_bay_area_long_format_plus_cross_bay_col.csv'
```

Comment:
To make your code work for other people, use a relative path instead of an absolute path.
```suggestion
     PATH = '../../data/raw/spring_2016_all_bay_area_long_format_plus_cross_bay_col.csv'
```
Don't mind my terrible whitespacing. Github editing was a struggle!